File: typescript/src/array/insert-interval.ts
function insert(intervals: number[][], newInterval: number[]): number[][] {
  let result: number[][] = [];

  for (let i = 0; i < intervals.length; i++) {
    if (newInterval[1] < intervals[i][0]) {
      result.push(newInterval);
      intervals.slice(i).forEach(interval => result.push(interval));
      return result;
    } else if (newInterval[0] > intervals[i][1]) {
      result.push(intervals[i]);
    } else {
      newInterval = [
        Math.min(newInterval[0], intervals[i][0]),
        Math.max(newInterval[1], intervals[i][1])
      ];
    }
  }

  result.push(newInterval);

  return result;
};

File: cs/LeetCode/LeetCode/Array/_InsertInterval.cs
namespace LeetCode.Array
{
    internal class _InsertInterval
    {
        public int[][] Insert(int[][] intervals, int[] newInterval)
        {
            var result = new List<int[]>();

            foreach (var interval in intervals)
            {
                if (interval[0] > newInterval[1])
                {
                    result.Add(newInterval);
                    newInterval = interval;
                }
                else if (interval[1] < newInterval[0])
                {
                    result.Add(interval);
                }
                else
                {
                    newInterval[0] = Math.Min(interval[0], newInterval[0]);
                    newInterval[1] = Math.Max(interval[1], newInterval[1]);
                }
            }

            result.Add(newInterval);
            return result.ToArray();
        }
    }
}
